craft_application/models/project.py

```diff
@@ -210,9 +212,9 @@ def _warn_deprecation(self) -> Self:
 
     @pydantic.field_validator("platforms", mode="before")
     @classmethod
-    def _populate_platforms(cls, platforms: dict[str, Any]) -> dict[str, Any]:
-        """Populate empty platform entries."""
-        return _populate_platforms(platforms)
+    def _expand_shorthand_platforms(cls, platforms: dict[str, Any]) -> dict[str, Any]:
```

Comment:
I'm assuming it's safe to rename internal validators? The name is misleading.

Reply:
I don't like the old name, but renaming protected methods is an API change since they can be overridden, so we can only do this on a major release.

Reply:
Wishful thinking, reverted.